frame/airdrop/src/lib.rs

```diff
@@ -0,0 +1,977 @@
+
+			match Associations::<T>::get(airdrop_id, reward_account.clone()) {
+				// Confirm association matches
+				Some(associated_account) => {
+					ensure!(associated_account == identity, Error::<T>::InvalidProof);
+				},
+				// If no association exist, create a new one
+				None => {
+					Associations::<T>::insert(airdrop_id, reward_account.clone(), identity.clone());
+				},
+			}
+
+			<Self as Airdropper>::claim(airdrop_id, identity, reward_account)
+		}
+	}
+
+	#[pallet::extra_constants]
+	impl<T: Config> Pallet<T> {
+		/// The AccountId of this pallet.
+		pub fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account()
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Gets the account ID to be used by the Airdrop.
+		pub(crate) fn get_airdrop_account_id(airdrop_id: T::AirdropId) -> AccountIdOf<T> {
+			T::PalletId::get().into_sub_account(airdrop_id)
+		}
+
+		/// Gets the [`Airdrop`](crate::models::Airdrop) associated with the `airdrop_id`
+		///
+		/// # Errors
+		/// * `AirdropDoesNotExist` - No Airdrop exist that is associated 'airdrop_id'
+		pub(crate) fn get_airdrop(airdrop_id: &T::AirdropId) -> Result<AirdropOf<T>, Error<T>> {
+			Airdrops::<T>::try_get(airdrop_id).map_err(|_| Error::<T>::AirdropDoesNotExist)
+		}
+
+		/// Calculates the current [`AirdropState`](crate::models::AirdropState) of an Airdrop
+		///
+		/// # Errors
+		/// * `AirdropDoesNotExist` - No Airdrop exist that is associated 'airdrop_id'
+		pub(crate) fn get_airdrop_state(
+			airdrop_id: T::AirdropId,
+		) -> Result<AirdropState, Error<T>> {
+			let airdrop = Self::get_airdrop(&airdrop_id)?;
+
+			if airdrop.disabled {
+				return Ok(AirdropState::Disabled)
+			}
+
+			airdrop.start.map_or(Ok(AirdropState::Created), |start| {
+				if start <= T::Time::now() {
+					Ok(AirdropState::Enabled)
+				} else {
+					Ok(AirdropState::Created)
+				}
+			})
+		}
+
+		/// Gets the [`RecipientFund`](crate::models::RecipientFund) of an Airdrop that is
+		/// associated with the `identity`.
+		///
+		/// # Errors
+		/// * `RecipientNotFound` - No recipient associated with the `identity` could be found.
+		pub(crate) fn get_recipient_fund(
+			airdrop_id: T::AirdropId,
+			identity: IdentityOf<T>,
+		) -> Result<RecipientFundOf<T>, Error<T>> {
+			RecipientFunds::<T>::try_get(airdrop_id, identity)
+				.map_err(|_| Error::<T>::RecipientNotFound)
+		}
+
+		/// Gets the remote account address from the `Proof`.
+		///
+		/// # Errors
+		/// * `InvalidProof` - If the proof is invalid, an error will be returned.
+		pub(crate) fn get_identity(
+			proof: ProofOf<T>,
+			reward_account: &<T as frame_system::Config>::AccountId,
+			prefix: &[u8],
+		) -> Result<IdentityOf<T>, DispatchErrorWithPostInfo<PostDispatchInfo>> {
+			let identity = match proof {
+				Proof::Ethereum(eth_proof) => {
+					let reward_account_encoded =
+						reward_account.using_encoded(|x| hex::encode(x).as_bytes().to_vec());
```

Comment:
this `.using_encoded(|x| hex::encode(x).as_bytes().to_vec())` pattern is used a lot, maybe extract it to a helper trait in composable_support?

Reply:
06ca34c8dfeb1633c276f7640d567147afeb4f27